### PR TITLE
configs: add Fedora ELN support

### PR DESCRIPTION
**Pull request by praiskup**
Fixes: #614

### CODE REVIEW

**Comment by praiskup** [comment]
Some notes:
- we sill need all the architectures before merging
- probably we need to define %dist/%eln to something reasonable
- do we need to set module_platform_id to correct value?
- gpg key is the rawhide key

File: mock-core-configs/etc/mock/templates/fedora-eln.tpl [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+gpgcheck=0
+assumeyes=1
+syslog_ident=mock
+syslog_device=
+install_weak_deps=0
+metadata_expire=0
+best=1
+
+# TODO
+module_platform_id=platform:eln
```
Comment (praiskup) [inline mock-core-configs/etc/mock/templates/fedora-eln.tpl:27]: @sgallagher is this correct?

**Comment by praiskup** [comment]
@sgallagher Or can we expect that minimal buildroot will define %dist in future?
It is usually defined in `/usr/lib/rpm/macros.d/macros.dist`.

File: mock-core-configs/etc/mock/templates/fedora-eln.tpl (hunk 1 above)
Comment (sgallagher) [reply mock-core-configs/etc/mock/templates/fedora-eln.tpl:27]: The modularity side of things is still being figured out. Let me get back to you on that tomorrow, please.

**Comment by sgallagher** [comment]
> @sgallagher Or can we expect that minimal buildroot will define %dist in future?
> It is usually defined in `/usr/lib/rpm/macros.d/macros.dist` (inside buildroot).

@tdawson is working on that this week. That said, there's no *harm* in defining `%dist` in the meantime.

File: mock-core-configs/etc/mock/templates/fedora-eln.tpl (hunk 1 above)
Comment (praiskup) [reply mock-core-configs/etc/mock/templates/fedora-eln.tpl:27]: Gently ping here.

File: mock-core-configs/etc/mock/templates/fedora-eln.tpl [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+name=local
+baseurl=https://kojipkgs.fedoraproject.org/repos/eln-build/latest/$basearch/
+cost=2000
+enabled=0
+skip_if_unavailable=False
+
+[eln]
+name=Fedora ELN - Developmental modular packages for the next Enterprise Linux release
+#metalink=https://mirrors.fedoraproject.org/metalink?repo=eln&arch=$basearch
+baseurl=https://odcs.fedoraproject.org/composes/production/latest-Fedora-ELN/compose/Everything/$basearch/os/
```
Comment (praiskup) [inline mock-core-configs/etc/mock/templates/fedora-eln.tpl:41]: @sgallagher I'm not able to install dnf from here.  Is this just a Rawhide addon (meaning that we are missing rawhide repo here)?

Comment (sgallagher) [reply mock-core-configs/etc/mock/templates/fedora-eln.tpl:41]: https://odcs.fedoraproject.org/composes/production/latest-Fedora-ELN/compose/Everything/x86_64/os/Packages/d/dnf-4.2.23-2.eln103.noarch.rpm exists, so I'm not sure what you mean.

Is it missing a dependency?

Comment (praiskup) [reply mock-core-configs/etc/mock/templates/fedora-eln.tpl:41]: > Is it missing a dependency?

Yes:
```
Error: 
 Problem 1: package dnf-4.2.23-2.eln103.noarch requires python3-dnf = 4.2.23-2.eln103, but none of the providers can be installed
  - package python3-dnf-4.2.23-2.eln103.noarch requires dnf-data = 4.2.23-2.eln103, but none of the providers can be installed
  - conflicting requests
  - nothing provides libreport-filesystem needed by dnf-data-4.2.23-2.eln103.noarch
 Problem 2: package python3-dnf-plugins-core-4.0.16-4.eln103.noarch requires python3-dnf >= 4.2.22, but none of the providers can be installed
  - package dnf-plugins-core-4.0.16-4.eln103.noarch requires python3-dnf-plugins-core = 4.0.16-4.eln103, but none of the providers can be installed
  - package python3-dnf-4.2.23-2.eln103.noarch requires dnf-data = 4.2.23-2.eln103, but none of the providers can be installed
  - conflicting requests
  - nothing provides libreport-filesystem needed by dnf-data-4.2.23-2.eln103.noarch
(try to add '--skip-broken' to skip uninstallable packages or '--nobest' to use not only best candidate packages)
```

Comment (praiskup) [reply mock-core-configs/etc/mock/templates/fedora-eln.tpl:41]: ping

Comment (mmathesius) [reply mock-core-configs/etc/mock/templates/fedora-eln.tpl:41]: The situation with missing dependencies is better now and `dnf` installs.

Now mock init fails with:
```
Module or Group 'build' is not available.
```

Comment (mmathesius) [reply mock-core-configs/etc/mock/templates/fedora-eln.tpl:41]: It appears `fedora-eln.tpl` just needs a slight fix to work:
```
-config_opts['chroot_setup_cmd'] = 'install @build'
+config_opts['chroot_setup_cmd'] = 'install @buildsys-build'
```

File: mock-core-configs/etc/mock/templates/fedora-eln.tpl [hunk 3]
```diff
@@ -0,0 +1,70 @@
+config_opts['root'] = 'fedora-eln-{{ target_arch }}'
+
+config_opts['chroot_setup_cmd'] = 'install @build'
```
Comment (mmathesius) [inline mock-core-configs/etc/mock/templates/fedora-eln.tpl:3]:
```suggestion
config_opts['chroot_setup_cmd'] = 'install @buildsys-build'
```

File: mock-core-configs/etc/mock/templates/fedora-eln.tpl (hunk 2 above)
Comment (praiskup) [reply mock-core-configs/etc/mock/templates/fedora-eln.tpl:41]: Indeed, it just works now.  Thank you for the ping.

**Comment by praiskup** [comment]
Does anyone want to take a final look? (seems ready to merge)

**Comment by mmathesius** [comment]
> Does anyone want to take a final look? (seems ready to merge)

Your latest update works for me for `x86_64` and `i686` (on my `x86_64` workstation). Thanks!

**Comment by praiskup** [comment]
thanks

**Comment by hroncok** [comment]
Thank You for this!

**Comment by mmathesius** [comment]
@praiskup Is there an ETA when this might make it into a release of the `mock-core-configs` package?

**Comment by praiskup** [comment]
I am able to wrap a new release right now (we have to release a new mock.rpm, too, though).  @Conan-Kudo how it goes with the new openmandriva configs?  Should we wait with the release?

**Comment by praiskup** [comment]
s/right now/this week/ (we need to run the testsuite, and it eats some time)

**Comment by mmathesius** [comment]
Thank you for the update.

If it makes things any quicker or easier, I'd be happy to submit a release bump PR to the Fedora package that simply pulls in this commit as a patch.

**Pull request merged**